Library/Homebrew/test/support/helper/cmd/brew-verify-undefined.rb

```diff
@@ -0,0 +1,33 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "cli/parser"
+
+UNDEFINED_CONSTANTS = %w[
+  Cask::Cask
+  Formula
+  Formulary
+  Homebrew::API
+  Tap
+].freeze
+
+parser = Homebrew::CLI::Parser.new do
```

Comment:
Note that https://github.com/Homebrew/brew/pull/18008 will require passing an `AbstractCommand` class when instantiating a `Parser`. (I'd like Parser to have a class-level `parse` method, but that's a massive refactor…)

Reply:
@dduugg The plan is to wait until your PR gets merged in before rebasing this one. I didn't realize there a conflict before I opened it and there's no rush here. My hope is that the main logic shouldn't have to change too much.